components/raftstore-v2/src/operation/command/admin/flashback.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_traits::{KvEngine, RaftEngine, RaftLogBatch};
+use fail::fail_point;
+use kvproto::{
+    raft_cmdpb::{AdminCmdType, AdminRequest, AdminResponse, RaftCmdRequest},
+    raft_serverpb::RegionLocalState,
+};
+use protobuf::Message;
+use raftstore::{coprocessor::RegionChangeReason, store::metrics::PEER_ADMIN_CMD_COUNTER, Result};
+
+use super::AdminCmdResult;
+use crate::{
+    batch::StoreContext,
+    fsm::ApplyResReporter,
+    raft::{Apply, Peer},
+};
+
+#[derive(Debug)]
+pub struct SetFlashbackState {
+    index: u64,
+    region_state: RegionLocalState,
+}
+
+impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
+    pub fn propose_flashback<T>(
+        &mut self,
+        store_ctx: &mut StoreContext<EK, ER, T>,
+        req: RaftCmdRequest,
+    ) -> Result<u64> {
+        let data = req.write_to_bytes().unwrap();
+        self.propose(store_ctx, data)
+    }
+}
+
+impl<EK: KvEngine, R: ApplyResReporter> Apply<EK, R> {
+    pub fn apply_flashback(
+        &mut self,
+        index: u64,
+        req: &AdminRequest,
+    ) -> Result<(AdminResponse, AdminCmdResult)> {
+        let is_in_flashback = req.get_cmd_type() == AdminCmdType::PrepareFlashback;
```

Comment:
Since the fn is apply_flashback,  req.get_cmd_type() must be AdminCmdType::PrepareFlashback. 
It can use an assert to assure it.